### PR TITLE
container: Remove VFIO lazy attach handling

### LINKED ISSUE #3108: virtcontainers/containerd.go depends on qemu
`Container.create` implements a hard coded lazy attach for VFIO devices with IO space larger than 4G. The hard coding is specific to both qemu and the `q35` machine type.

With @dgibson work for a proper VFIO and ACPI based hotplug support, I don't think we need this hard coded corner case any more.

/cc @Jimmy-Xu WDYT?

### PR DESCRIPTION
**Pull request by sameo**
With the recently added VFIO fixes and support, we should not need that
anymore.

Fixes #3108

Signed-off-by: Samuel Ortiz <s.ortiz@apple.com>

### CODE REVIEW

**Comment by devimc** [comment]
@Jimmy-Xu please take a look

**Review by devimc — DISMISSED** [review]
thanks @sameo - I think so, lazy attack is no longer needed

**Review by dgibson — APPROVED** [review]
LGTM, however I do not have the equipment to actually verify that this doesn't break anything.

**Comment by Jimmy-Xu** [comment]
> 

@devimc  @sameo  Cool. I will test this PR.

File: src/runtime/virtcontainers/container.go
Comment (jodh-intel) [inline src/runtime/virtcontainers/container.go:893]: How about logging the error in a debug call here?

**Review by jodh-intel — APPROVED** [review]
Nice clean up! Deleting all this code must have been great fun @sameo ! ;)

**Comment by fidencio** [comment]
@Jimmy-Xu, did you happen to have some time to test this one?

**Comment by jodh-intel** [comment]
@sameo - please can you resolve the conflict.

@Jimmy-Xu - have you had a chance to tes this?

**Comment by jodh-intel** [comment]
Ping @sameo.

**Comment by fidencio** [comment]
A head's up that may affect your PR.

In the Architecture Committee meeting from January 25th, 2022, the Architecture Committee has agreed on using the "Dismiss stale pull request approvals when new commits are pushed" configuration from GitHub.  It basically means that if your PR has been rebased or updated, the approvals given will be erased.

In order to minimize traumas due to the new approach, please, consider adding a note on the changes done before the rebase / force-push, and also pinging the reviewers for a subsequent round of reviews.

Thanks for your understanding!

Related issue: https://github.com/kata-containers/community/issues/249

**Comment by sameo** [comment]
PR is rebased now.

FWIW, without this PR `container.go` only builds on Linux. This is why I added the `area/darwin` label.

**Comment by sameo** [comment]
@dgibson @devimc @Jimmy-Xu @jodh-intel Would you mind re-reviewing/approving?

**Comment by egernst** [comment]
@Jimmy-Xu are you able to test?

**Review by jodh-intel — APPROVED** [review]
Thanks @sameo.

lgtm

**Comment by sameo** [comment]
Folks, I am pinging @Jimmy-Xu on Slack but could we consider merging that PR? It's been sitting there for a long time already...

**Comment by fidencio** [comment]
> Folks, I am pinging @Jimmy-Xu on Slack but could we consider merging that PR? It's been sitting there for a long time already...

What about getting it merged on Monday if there's no response from @Jimmy-Xu till there?

**Comment by Jimmy-Xu** [comment]
@sameo  @fidencio 
I don't have a test environment now. 
I think this PR can be merged if there is no conflict between PCI rescan and large BAR space VFIO device attachment.
So it's LGTM.
Once I have a test environment, I will do a test and give confirmation.

**Review by dgibson — APPROVED** [review]
Still LGTM.

**Pull request merged**
